fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguagesPreferenceFragmentIntegration.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations.preferences.downloadlanguages
+
+import android.content.Context
+import android.os.PowerManager
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.fenix.wifi.WifiConnectionMonitor
+
+/**
+ * Helper for observing wifiConnection and power saving mode.
+ */
+class DownloadLanguagesPreferenceFragmentIntegration(
+    context: Context,
+    private val wifiConnectionMonitor: WifiConnectionMonitor,
+    private val savingModeListener: (Boolean) -> Unit,
+) : LifecycleAwareFeature {
+
+    private val powerManager =
```

Comment:
Could you give more information or background on this call?

To determine when we should show the download warning, we should be checking for [Data Saver](https://source.android.com/docs/core/data/data-saver) mode and if the phone is using mobile data.

We should be able to check Data Saver [using this API](https://developer.android.com/develop/connectivity/network-ops/data-saver#status). 

I think we should consider data saver enabled when either `RESTRICT_BACKGROUND_STATUS_ENABLED` OR  `RESTRICT_BACKGROUND_STATUS_WHITELISTED` is returned.

For the WiFi criteria, we might need to do a little research if the criteria should be the negative of "not WiFi" or the positive of "is mobile data".

Also, if we have a way to mock the network status and data saver mode easily, I'd really prefer a test of this logic.

Reply:
Done, I modify the class and I added unit tests.

Reply:
Awesome, thanks!